internal/pkg/status/legacy_status.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clientu "sigs.k8s.io/cli-experimental/internal/pkg/client/unstructured"
+)
+
+// IsReadyFn - status getter
+type IsReadyFn func(*unstructured.Unstructured) (bool, error)
+
+var legacyTypes = map[string]map[string]IsReadyFn{
+	"": map[string]IsReadyFn{
+		"Service":               alwaysReady,
+		"Pod":                   podReady,
+		"PersistentVolumeClaim": pvcReady,
+	},
+	"apps": map[string]IsReadyFn{
+		"StatefulSet": stsReady,
+		"DaemonSet":   daemonsetReady,
+		"Deployment":  deploymentReady,
+		"ReplicaSet":  replicasetReady,
+	},
+	"policy": map[string]IsReadyFn{
+		"PodDisruptionBudget": pdbReady,
+	},
+	"batch": map[string]IsReadyFn{
+		"CronJob": cronjobReady,
+		"Job":     jobReady,
+	},
+}
+
+// GetLegacyReadyFn - True if we handle it as a known type
+func GetLegacyReadyFn(u *unstructured.Unstructured) IsReadyFn {
+	gvk := u.GroupVersionKind()
+	g := gvk.Group
+	k := gvk.Kind
+	if _, ok := legacyTypes[g]; ok {
```

Comment:
why not just key off of `GroupKind`?

Reply:
Can do. Is it optimal ?

Reply:
It doesn't require a double lookup, so it's a bit easier to ready

Reply:
Made the changes. Its a simple map now.